### ISSUE TITLE
Call next method  to start DoubleFlutterActivity after finish SingleFlutterActivity, and call next in the new widget, app crash

### ISSUE THREAD

**Issue by heijianye** [flutter/flutter#146561]
### Steps to reproduce

```
1.add close method in Multi_Flutter project, and call finish method in close method;
interface EngineBindingsDelegate {
    fun onNext(value: HashMap<String, String>)
    fun close()
}

override fun close() {
        finish()
    }

2.call close method in dart click method, and call next after that
_channel.invokeMethod<void>("close").then((value) => _channel
                    .invokeMethod<void>("next"));

3.the new FLutterActivity will show normally, and call next method in the new dart codes:
_channel.invokeMethod<void>("next");

```
4.the crash happened.
`2024-04-10 18:50:36.098 22493-22493 libc                    dev.flutter.multipleflutters         A  Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x8 in tid 22493 (ultipleflutters), pid 22493 (ultipleflutters)`

### Expected results

App not crashed after the new FlutterActivity onCreate.

### Actual results

App show black screen and crash after new FlutterActivity start.

### Code sample

<details open><summary>Code sample</summary>

```dart
_channel.invokeMethod<void>("close").then((value) => _channel
                    .invokeMethod<void>("next"));


_channel.invokeMethod<void>("next");

```

、、、kotlin
interface EngineBindingsDelegate {
    fun onNext(value: HashMap<String, String>)
    fun close()

}

override fun close() {
        finish()
    }
、、、

demo zip file:

[default.zip](https://github.com/flutter/flutter/files/14931477/default.zip)

</details>

### Screenshots or Video

ScreenRecord when issue happened:
https://github.com/flutter/flutter/assets/13232266/957d7d18-89eb-4d61-abb0-b09e83a8437c

### Logs

And the following is the full trace stack log.
[test.txt](https://github.com/flutter/flutter/files/14931195/test.txt)

### Flutter Doctor output

<details open><summary>Doctor output</summary>

multiple_flutters % flutter doctor -v
[!] Flutter (Channel dev, 3.7.12, on macOS 14.3.1 23D60 darwin-x64, locale zh-Hans-CN)
    • Flutter version 3.7.12 on channel dev at /Users/heijianye/flutter_flutter
    ! Upstream repository https://gitee.com/openharmony-sig/flutter_flutter.git is not a standard remote.
      Set environment variable "FLUTTER_GIT_URL" to https://gitee.com/openharmony-sig/flutter_flutter.git to dismiss this error.
    • Framework revision deae5c30fb (4 weeks ago), 2024-03-13 10:28:57 +0000
    • Engine revision 1a65d409c7
    • Dart version 2.19.6
    • DevTools version 2.20.1
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] HarmonyOS toolchain - develop for HarmonyOS devices
    • OpenHarmony Sdk location: /Users/heijianye/Library/Huawei/Sdk, available api versions has [HarmonyOS-NEXT-DP1, HarmonyOS-NEXT-DP0]

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/heijianye/Library/Android/sdk
    • Platform android-34, build-tools 33.0.0
    • ANDROID_HOME = /Users/heijianye/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15E204a
    • CocoaPods version 1.13.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.87.0)
    • VS Code at /Users/heijianye/files/压缩包和安装包/Visual Studio Code.app/Contents
    • Flutter extension version 3.86.0

[✓] Connected device (3 available)
    • PEDM00 (mobile) • 2395c374 • android-arm64  • Android 12 (API 31)
    • macOS (desktop) • macos    • darwin-x64     • macOS 14.3.1 23D60 darwin-x64
    • Chrome (web)    • chrome   • web-javascript • Google Chrome 123.0.6312.107

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

</details>

**Comment by heijianye** [comment]
Sometimes, the app do not exit, but the next can not still open new engine, and the issue is same as following:
2024-04-10 19:33:31.880  9774-9774  libc                    dev.flutter.multipleflutters         A  Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x8 in tid 9774 (ultipleflutters), pid 9774 (ultipleflutters)

[new_test.txt](https://github.com/flutter/flutter/files/14931542/new_test.txt)
 
video as following:
https://github.com/flutter/flutter/assets/13232266/d1844708-d2d0-4f5c-aedd-2807c8532a98

**Comment by heijianye** [comment]
Hi @darshankawar，
I check with latest 3.19.5 version, the issue does not happen with the same codes for multi-flutters.

But I have got a question, if I have to use 3.7.12 in in some projects temporarily. What should I do to overcome this issue?

And according to the crash error, can we confirm how the latest flutter resolved the issue?

**Comment by darshankawar** [comment]
> But I have got a question, if I have to use 3.7.12 in in some projects temporarily. What should I do to overcome this issue?
> 
> And according to the crash error, can we confirm how the latest flutter resolved the issue?

It will be hard to pinpoint which fixes resolved the issue because 3.7.12 is very old version and there have been lot of updates made to the framework since then until current version. I suggest you to be on latest version to get best out of all the features provided by the sdk instead of being on 3.7.12.

Since the issue / crash doesn't occur on latest version per your comment, I'll go ahead and close the issue as resolved. You may go through the documentation like this https://docs.flutter.dev/add-to-app/android/add-flutter-screen for your further reference.
